doc/content/reference/id-eui-constraints/index.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "ID and EUI Constraints"
+description: ""
+---
+
+This reference explains how {{% tts %}} deals with IDs and EUIs, including regular expression requirements, uniqueness, and how deletion is handled.
+
+<!--more-->
+
+## Requirements of an ID or EUI
+
+An ID or EUI in {{% tts %}} must:
+
+- Have a length of between 3 and 36 characters (inclusive)
+- Consist of lowercase letters, numbers, and non-consecutive dashes
+- **NOT** begin or end with a dash
+
+The following regular expression is used to validate IDs and EUIs:
+
+`(^[a-z0-9](?:[-]?[a-z0-9]){2,}$)`
+
+## Uniqueness
+
+Within a deployment, there can only be:
+
+- One tenant with the same ID
+- One gateway with the same EUI (across all tenants)
+- One end device with the same DevEUI,JoinEUI tuple (across all tenants)
+
+Within a tenant, there can only be:
+
+- One user or organization with the same ID
+- One application with the same ID
+- One gateway with the same ID
+- One OAuth client with the same ID
+
+Within an application, there can only be:
+
+- One end device with the same ID
+
+## Deleted Entities
+
+When deleting entities, some IDs are retained so that they can't be registered again. This is for security reasons as integrations may store historical data by ID, and you don't want that someone else can register an ID and get your historical data.
+
+EUIs are released in some cases so that they can be registered in other applications or tenants.
+
+- When a user or organization is deleted, the ID is retained and cannot be reused within the deployment
```

Comment:
I think we already covered the "within the xxx" above. Otherwise this is incorrect, because the same user/org IDs can be used in different tenants.

Reply:
Same story below.

Reply:
Okay. I'm going to leave the redundant but factually correct statements just to reiterate that the deleted IDs are retained and can't be used in the tenant, if that's correct:

- When a user or organization is deleted, the ID is retained and cannot be reused within the tenant
- When an application is deleted, the ID is retained and cannot be reused within the tenant
- When a gateway is deleted, the ID is retained but the gateway EUI can be reused
- When an end device is deleted, the ID and EUI are released and can be reused